src/main/scala/mesosphere/marathon/MarathonSchedulerActor.scala

```diff
@@ -271,7 +272,7 @@ class MarathonSchedulerActor(
     withLockFor(Set(appId))(f)
 
   // there has to be a better way...
-  def driver: SchedulerDriver = MarathonSchedulerDriver.driver.get
+  def driver: SchedulerDriver = marathonSchedulerDriverHolder.driver.get
```

Comment:
Usually, you would just wire in the driver into all classes needing them. Unfortunately, the driver is recreated/destroyed whenever leadership changes. Thus, you cannot wire it in directly.

The best way would be to tear down all state/actors when we abdicate leadership and recreate them when we become leader. Then we could just wire the driver in directly. That would require a bigger set of changes, unfortunately.